hcloud/schema/error_test.go

```diff
@@ -0,0 +1,54 @@
+package schema
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestError(t *testing.T) {
+	t.Run("UnmarshalJSON", func(t *testing.T) {
+		data := []byte(`{
+			"code": "invalid_input",
+			"message": "invalid input",
+			"details": {
+				"fields": [
+					{
+						"name": "broken_field",
+						"messages": ["is required"]
+					}
+				]
+			}
+		}`)
+
+		e := &Error{}
+		err := json.Unmarshal(data, e)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if e.Code != "invalid_input" {
+			t.Errorf("unexpected Code: %v", e.Code)
+		}
+		if e.Message != "invalid input" {
+			t.Errorf("unexpected Message: %v", e.Message)
+		}
+		if e.Details == nil {
+			t.Fatalf("unexpected Details: %v", e.Details)
+		}
+		if d, ok := e.Details.(*ErrorInvalidInputDetails); !ok {
+			t.Fatalf("unexpected Details type (should be *ErrorInvalidInputDetails): %v", e.Details)
+		} else {
```

Comment:
`else` block not needed since `t.Fatalf` will stop execution.

Reply:
the else block was needed because `d` is scoped to the `if` block